### PR TITLE
dev-python/prov: new ebuild

### PR DESCRIPTION
**Pull request by TheChymera**
Package-Manager: Portage-2.3.5, Repoman-2.3.2

@SoapGentoo @gktrk

### CODE REVIEW

File: dev-python/prov/prov-1.5.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_4,3_5} )
```
Comment (SoapGentoo) [inline dev-python/prov/prov-1.5.0.ebuild]: please also try 3.6

File: dev-python/prov/prov-1.5.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	"
+RDEPEND="dev-python/python-dateutil[${PYTHON_USEDEP}]
+	dev-python/lxml[${PYTHON_USEDEP}]
+	>=dev-python/networkx-1.10[${PYTHON_USEDEP}]
+	>=dev-python/six-1.10[${PYTHON_USEDEP}]
+	"
+
+python_test() {
+	${EPYTHON} -m unittest discover || die
```
Comment (SoapGentoo) [inline dev-python/prov/prov-1.5.0.ebuild]: `"${EPYTHON}" -m unittest discover || die` (notice the quotes)

File: dev-python/prov/prov-1.5.0.ebuild (hunk 1 above)
Comment (TheChymera) [inline dev-python/prov/prov-1.5.0.ebuild]: I am not using 3.6 yet on any of my machines

Comment (SoapGentoo) [inline dev-python/prov/prov-1.5.0.ebuild]: could you try?

**Comment by SoapGentoo** [comment]
@TheChymera always squash your commits into one

File: dev-python/prov/prov-1.5.0.ebuild (hunk 1 above)
Comment (TheChymera) [inline dev-python/prov/prov-1.5.0.ebuild]: How do I get python3.6 Also, I think eselect is confused, how do I make it understand that I actually don't have python3.5?

```
zenbookhost ~ # eix -e python
[U] dev-lang/python
     Available versions:  
     (2.7)  2.7.12 (~)2.7.13
     (3.4)  3.4.5(3.4/3.4m) (~)3.4.6(3.4/3.4m)
     (3.5)  (~)3.5.2(3.5/3.5m) (~)3.5.3(3.5/3.5m)
     (3.6)  (~)3.6.0(3.6/3.6m) (~)3.6.1(3.6/3.6m)
       {-berkdb build doc examples gdbm hardened ipv6 libressl +ncurses +readline sqlite +ssl +threads tk +wide-unicode wininst +xml ELIBC="uclibc"}
     Installed versions:  2.7.13(2.7)(11:22:28 21.02.2017)(gdbm ipv6 ncurses readline sqlite ssl threads wide-unicode xml -berkdb -build -doc -examples -hardened -libressl -tk -wininst ELIBC="-uclibc") 3.4.6(3.4)(11:25:20 21.02.2017)(gdbm ipv6 ncurses readline sqlite ssl threads xml -build -examples -hardened -libressl -tk -wininst ELIBC="-uclibc")
     Homepage:            https://www.python.org/
     Description:         An interpreted, interactive, object-oriented programming language

zenbookhost ~ # eselect python list
Available Python interpreters, in order of preference:
  [1]   python2.7
  [2]   python3.4
  [3]   python3.5
zenbookhost ~ # python3.5
-su: python3.5: command not found
```

**Comment by gktrk** [comment]
Did you check the test dependencies for this package?

```
>>> Test phase: dev-python/prov-1.5.0
...
======================================================================
ERROR: test_serialize_to_path (prov.tests.test_extras.TestExtras)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "prov/tests/test_extras.py", line 145, in test_serialize_to_path
    document.serialize("output.json")
  File "prov/model.py", line 1668, in serialize
    serializer = serializers.get(format)(self)
  File "prov/serializers/__init__.py", line 57, in get
    Registry.load_serializers()
  File "prov/serializers/__init__.py", line 41, in load_serializers
    from prov.serializers.provrdf import ProvRDFSerializer
  File "prov/serializers/provrdf.py", line 31, in <module>
    from rdflib.term import URIRef, BNode
ImportError: No module named rdflib.term

======================================================================
ERROR: dev.test_count_Streaming (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: dev.test_count_Streaming
Traceback (most recent call last):
  File "/usr/lib64/python2.7/unittest/loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "/usr/lib64/python2.7/unittest/loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "dev/test_count_Streaming.py", line 4, in <module>
    from yajl import YajlParser, YajlContentHandler
ImportError: No module named yajl
```

**Comment by TheChymera** [comment]
@gktrk rdflib is indeed something I missed. The yajl error (which I could reproduce) looks to me like some problem with setuptools (as I expect that is the package that creates `/var/tmp/portage/dev-python/prov-1.5.0/work/prov-1.5.0/dev/test_count_Streaming.py`). Strange thing is: I cannot find the string yajl in either of the packages' source. Any idea what is going on here?

**Comment by gktrk** [comment]
> @gktrk rdflib is indeed something I missed. The yajl error (which I could reproduce) looks to me like some problem with setuptools (as I expect that is the package that creates /var/tmp/portage/dev-python/prov-1.5.0/work/prov-1.5.0/dev/test_count_Streaming.py). Strange thing is: I cannot find the string yajl in either of the packages' source. Any idea what is going on here?

I'm not quite sure myself either. In any case, yajl for python (https://github.com/pykler/yajl-py) is not in the tree. I've packaged it in my overlay just to see what happens (https://github.com/gktrk/kpanicd-overlay/blob/master/dev-python/yajl-py/yajl-py-2.1.1.ebuild) and the tests still failed:

```
ERROR: dev.test_count_Streaming (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: dev.test_count_Streaming
Traceback (most recent call last):
  File "/usr/lib64/python2.7/unittest/loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "/usr/lib64/python2.7/unittest/loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "dev/test_count_Streaming.py", line 7, in <module>
    @profile
NameError: name 'profile' is not defined


----------------------------------------------------------------------
Ran 957 tests in 54.576s

FAILED (errors=1, expected failures=17)
```

At this point, I'm ok with restricting the test phase.

**Comment by enku** [comment]
I think `@profile` may be coming from the use of the [line profiler](https://github.com/rkern/line_profiler).  At any rate, it's probably should not be something that is published.  Something one usually only used for debugging.  So I think that upstream should remove it from the code.

**Comment by TheChymera** [comment]
@gktrk can we pull now?

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/4e55823...0f2155f. Thanks!

I've unindented the closing `"` for `DEPEND` and `RDEPEND` prior to merging.

**Pull request closed without merging**